File: src/main/java/com/eager/questioncloud/domain/question/repository/QuestionJpaRepository.java
package com.eager.questioncloud.domain.question.repository;

import com.eager.questioncloud.domain.question.entity.QuestionEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface QuestionJpaRepository extends JpaRepository<QuestionEntity, Long> {
    Optional<QuestionEntity> findByIdAndCreatorId(Long id, Long creatorId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/infrastructure/repository/QuestionJpaRepository.kt
package com.eager.questioncloud.core.domain.question.infrastructure.repository;

import com.eager.questioncloud.core.domain.question.infrastructure.entity.QuestionEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface QuestionJpaRepository extends JpaRepository<QuestionEntity, Long> {
    Optional<QuestionEntity> findByIdAndCreatorId(Long id, Long creatorId);
}
